Minor contributing fixups

thaJeztah opened the pull request:
This is a follow-up of https://github.com/docker/distribution/pull/545 and makes some stylistic changes in the CONTRIBUTING document;

- Replaces checkboxes with regular bullets. Because the checkboxes are
  read-only for non-contributors, they are presented as 'greyed out',
  which makes them harder to see (depending on the browser used)
- Replaces bullets for numbered lists in the "creating an issue" and
  "contributing a patch" sections. The numbers make it easier for
  readers to track where they left off.
- Cleans up some minor Markdown changes for consistency

dmcgowan commented:
LGTM

dmp42 commented on CONTRIBUTING.md:
```diff
@@ -4,70 +4,68 @@
+ - any other [Docker Hub](https://hub.docker.com/) issue
 
-Then please do not report your issue here - you should instead report it to [https://support.docker.com]
+Then please do not report your issue here - you should instead report it to [https://support.docker.com](https://hub.docker.com/)
```
I don't understand.
Support is the correct address.
What am I missing?

stevvooe replied:
@dmp42 Looks like a mistake. @thaJeztah Does support.docker.com not work for you?

thaJeztah replied:
Huh, don't know what happened there? I only meant to make the link a proper Markdown link, must have had something else on my clipboard. 😊

Will fix. Thanks for spotting that!

dmp42 commented on CONTRIBUTING.md:
```diff
@@ -4,70 +4,68 @@
+ 1. create a new issue, with a succinct title that describes your issue:
+   - bad title: "It doesn't work with my docker"
+   - good title: "Private registry push fail: 400 error with E_INVALID_DIGEST"
+ 2. copy the output of `docker version`, `docker info`, and your Registry version
```
I don't know about this. I like your compacter form, but people might just miss the last two

thaJeztah replied:
Yes I was a bit in doubt about this as well; my goal was to keep the overall steps down; perhaps I should make that a "sub" list (bullets) as well and see if that works

What's the right approach to get the "version" of registry? then I'll add that as well

dmp42 commented on CONTRIBUTING.md:
```diff
@@ -4,70 +4,68 @@
+ 3. copy the command line you used to launch your Registry
+ 4. restart your docker daemon in debug mode (add `-D` to the daemon launch arguments)
+ 5. reproduce your problem and get your docker daemon logs showing the error
+     - if relevant, copy your registry logs that show the error
```
This one is not a sub item of 5 - it's a separate item

dmp42 commented:
A couple of nits, but this is good. Thanks @thaJeztah !

thaJeztah commented:
@dmp42 thanks for reviewing. I updated the PR addressing your comments, most notably;

- Fixed the support.docker.com link (oops)
- Change the "copy" information to be a sub-bulleted listed (https://github.com/docker/distribution/pull/589#discussion_r31595945), see if you like it that way. (If not, and you prefer these steps to be in the numbered list itself, let me know)

dmp42 commented:
Thanks a lot for this @thaJeztah 
That's LGTM.

About getting the registry version out... mmm... I would probably exec into the container and run `registry --version`.

@stevvooe is there a simpler (out of container) way?

stevvooe commented:
@dmp42 This provides the version:

```
$ registry -version
registry docker/distribution v2.0.0-alpha.1+unknown
```

thaJeztah commented:
@stevvooe single dash?

Thinking what the easiest thing is to put there, given that people can run the registry in various setups.

Perhaps just `registry -version` and assume they understand they'll have to exec into a container if that's where it's running.

dmp42 commented:
I would rather assume the other way round: the default way to run the registry is from the container, and people running it otherwise would know what they do.

I'd suggest we go with `docker exec ID_OF_REGISTRY_IMAGE registry -version`.

thaJeztah commented:
@dmp42 updated, PTAL

dmp42 commented:
LGTM
We should give it a spin for a couple of weeks and see how that pans out.

@stevvooe merge at your will :)

stevvooe commented:
LGTM!
